pkg/cel/cel_test.go

```diff
@@ -0,0 +1,56 @@
+package cel
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestCelLibrary(t *testing.T) {
+	props := make([]map[string]interface{}, 1)
+	props[0] = map[string]interface{}{
+		"type":  "olm.test",
+		"value": "1.0.0",
+	}
+
+	propertiesMap := map[string]interface{}{"properties": props}
+
+	tests := []struct {
+		name  string
+		rule  string
+		out   bool
+		isErr bool
+	}{
```

Comment:
looks good. I would suggest having a check for each comparison direction in semver to really make sure it works, i.e.
in addition to testing `properties.exists(p, p.type == 'olm.test' && (semver_compare(p.value, '1.0.0') == 0))` add `properties.exists(p, p.type == 'olm.test' && (semver_compare(p.value, '0.9.0') > 0))` and `properties.exists(p, p.type == 'olm.test' && (semver_compare(p.value, '1.0.1') < 0))`

Reply:
it might also be worth exercising the bad semver code path

Reply:
also the non-boolean return type error code path

Reply:
Will see what I can do here.